crates/configuration/src/network.rs

```diff
@@ -52,27 +52,35 @@ states! {
 #[derive(Debug)]
 pub struct NetworkConfigBuilder<State> {
     config: NetworkConfig,
+    errors: Vec<Box<dyn Error>>,
     _state: PhantomData<State>,
 }
 
 impl Default for NetworkConfigBuilder<Initial> {
     fn default() -> Self {
         Self {
             config: NetworkConfig {
-                global_settings: GlobalSettingsBuilder::new().build(),
+                global_settings: GlobalSettingsBuilder::new().build().expect(
+                    "should have no errors for default builder. this is a bug, please report it",
```

Comment:
I think the convention is to have the error in lowercase (except for struct name or type), and I didn't put the if because this is a bug by definition, we don't expect this to happens, that's why we unwrap